### PR TITLE
Create facility enum

### PR DESCRIPTION
**Pull request by sven1103**
This PR adds a facility enum with five initial definitions.

### CODE REVIEW

**Comment by sven1103** [comment]
[#701: Create enumeration representing facilities in data-model-lib](https://trello.com/c/DJVy3tne/887-701-create-enumeration-representing-facilities-in-data-model-lib)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,11 +4,13 @@ Changelog
 
 This project adheres to `Semantic Versioning <https://semver.org/>`_.
 
-2.11.0-SNAPSHOT (2021-07-22)
+2.11.0-SNAPSHOT
 ----------------------------
 
 **Added**
 
+* A new enumeration for facilities ``life.qbic.datamodel.dtos.business.facilities.Facility``
```
Comment (KochTobi) [inline CHANGELOG.rst:12]:
```suggestion
* A new enumeration for facilities ``life.qbic.datamodel.dtos.business.facilities.Facility`` (`#244 <https://github.com/qbicsoftware/data-model-lib/pull/244>`_)
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * <p>Controlled vocabulary for concise core facility labels.</p>
+ *
+ * <p>The following facilities are currently defined:</p>
+ *
+ * <ul>
+ *     <li>CFMB: Core Facility for Medical Bioanalytics</li>
+ *     <li>IMGAG: Institute for Medical Genetics and Applied Genomics</li>
+ *     <li>MGM: Institute for Medical Microbiology and Hygiene</li>
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy:11]: Why is it `MGM` what does that stand for? Why not `MMH`or something similar? Isn't it https://uni-tuebingen.de/en/faculties/faculty-of-science/departments/interfaculty-facilities/imit/research-groups/medical-microbiology-and-hygiene/

File: src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ */
+enum Facility {
+
+    CFMB("Core Facility for Medical Bioanalytics"),
+    IMGAG("Institute for Medical Genetics and Applied Genomics"),
+    MGM("Institute for Medical Microbiology and Hygiene"),
+    QBIC("Quantitative Biology Center"),
+    PCT("Proteome Center Tübingen")
+
+    private final String value
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy:26]: I think we could change that to something meaningful. This way we can extend the enum by other properties if we need them in the future. And it is clear what is meant here.
```suggestion
    private final String fullName
```

**Review by KochTobi — CHANGES_REQUESTED** [review]
I have a small question to one of the enum fields and a suggestion regarding the property name

File: src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy (hunk 2 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy:11]: Thats the official acronym, see the offer footer please

File: src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    String getValue() {
+        return this.value
+    }
+
+    /**
+     * Returns a String representation of the enum item
+     * @return
+     */
+    @Override
+    String toString() {
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy:44]:
```suggestion
    Facility(String fullName) {
        this.fullName = fullName
    }
    /**
     * Returns the full name of the facility
     * @return the full name of the facility
     */
    String getFullName() {
        return this.fullName
    }

    /**
     * Returns a String representation of the enum item
     * @return
     */
    @Override
    String toString() {
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        return this.getValue()
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/facilities/Facility.groovy:45]:
```suggestion
        return this.getFullName()
```

**Review by KochTobi — DISMISSED** [review]

**Review by KochTobi — APPROVED** [review]

**Pull request merged**